### ISSUE TITLE
Add support for pseudo selectors and media queries

### ISSUE THREAD

**Issue by johanholmerin** [johanholmerin/style9#1]
Example from Facebook Workplace:

```javascript
{
  focused: {
    outline: "2px solid Highlight",
    "@media (-webkit-min-device-pixel-ratio: 0)": {
        outline: "5px auto -webkit-focus-ring-color"
    }
  },
  unfocused: {
    outline: "none"
  }
}

{
  typeaheadLayoutOpened: {
    borderBottomEndRadius: "8px",
    borderBottomStartRadius: "8px",
    "::before": {
      borderBottomEndRadius: "8px",
      borderBottomStartRadius: "8px"
    }
  }
}
```

**Comment by CanRau** [comment]
👍 
Something like customizable named keys would be cool as well like [fela-plugin-named-keys](https://github.com/robinweser/fela/tree/master/packages/fela-plugin-named-keys)

**Comment by johanholmerin** [comment]
Yes, this issue was about adding support for nesting, which is implemented in 0.5.0.
